control-plane/agents/src/bin/core/controller/scheduling/affinity_group.rs

```diff
@@ -61,6 +61,28 @@ pub(crate) async fn get_pool_ag_replica_count(
     pool_ag_replica_count
 }
 
+/// Get a set of nodes which contain the replica of a single-replica volume, part of the affinity group.
+pub(crate) async fn ag_restricted_nodes(
+    affinity_group_spec: &AffinityGroupSpec,
+    registry: &Registry,
+) -> HashSet<NodeId> {
+    let mut nodes = HashSet::<NodeId>::new();
+    let specs = registry.specs();
+    for volume_id in affinity_group_spec.volumes() {
+        let Some(volume) = registry.specs().volume_rsc(volume_id) else {
+            // should not really happen
+            continue;
+        };
+        if volume.lock().num_replicas != 1 {
+            continue;
+        }
+
+        tracing::info!("Adding {volume_id}");
```

Comment:
Can you add a bit more info on this log, just `Adding 9c39458e-5cbc-4ba2-bc2d-f040b79a8850` might look a bit vague?